### DISCUSSION TITLE
Changing profiles pre init doesn't work

### DISCUSSION THREAD

**Discussion by jelmerS2** [microsoft/MixedRealityToolkit-Unity#9154] [Q&A] [answered]
I'm trying to assign a profile to MRTK prior to initialization due to cross-platform development. 
There are several issues:
In [this guide](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Documentation/MixedRealityConfigurationGuide.md/#pre-mrtk-initialization-profile-switch), SetProfileBeforeInitialization() is mentioned. This functions seems to not exist anywhere. 

Following [this guide](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/MixedRealityConfigurationGuide.html#swapping-profiles-prior-to-mrtk-initialization), I run into the following errors at runtime: 

- There's already a IMixedRealityInputSystem.Mixed Reality Input System registered!
- Failed to start the Input System!
- There's already a IMixedRealityFocusProvider.Focus Provider registered!
- Failed to register the focus provider! The input system will not function without it.

My script is taken from there and as follows:

```
using Microsoft.MixedReality.Toolkit;
using UnityEditor;
using UnityEngine;
/// <summary>
/// Sample MonoBehaviour that will run before the MixedRealityToolkit object, and change
/// the profile, so that when the MRTK initializes it uses the profile specified below
/// rather than the one that is saved in its scene.
/// </summary>
/// <remarks>
/// Note that this script must have a higher priority in the script execution order compared
/// to that of MixedRealityToolkit.cs. See https://docs.unity3d.com/Manual/class-MonoManager.html
/// for more information on script execution order.
/// </remarks>
/// 
namespace Assets.Scripts.Core
{

    public class ProfileSwapper : MonoBehaviour
    {
        public MixedRealityToolkitConfigurationProfile arProfile;
        public MixedRealityToolkitConfigurationProfile vrProfile;


        void Awake()
        {
            MixedRealityToolkit.Instance.ActiveProfile = vrProfile;

        }
    }
}
```

Script exec order is: 
![Screenshot 2021-01-20 at 14 26 04](https://user-images.githubusercontent.com/47639577/105184203-5f931f80-5b2f-11eb-84e7-68c03e8fd0ad.png)

What am I doing wrong? Or is there a bug?

**Comment by keveleigh** [comment, marked as answer]
@MaxWang-MS made some fixes in https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8787 for the upcoming MRTK 2.6 release which should better support this scenario.

**Comment by MaxWang-MS** [comment]
There are known issues with profile changing in the current releases of MRTK. Like @keveleigh has mentioned, these issues have been fixed and will be included in the upcoming 2.6 release. You can try them out now by using the mrtk_development branch and the code there corresponds to [the first guide you mentioned](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Documentation/MixedRealityConfigurationGuide.md/#pre-mrtk-initialization-profile-switch).
